certauth.go

```diff
@@ -109,15 +114,57 @@ func (a *Auth) RouterHandler(h httprouter.Handle) httprouter.Handle {
 	return httprouter.Handle(func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 		// Let secure process the request. If it returns an error,
 		// that indicates the request should not continue.
-		if err := a.Process(w, r); err != nil {
+		var err error
+		if r, err = a.ProcessWithParams(w, r, ps); err != nil {
 			return
 		}
 
 		h(w, r, ps)
 	})
 }
 
-// ValidateRequest perfomrs verification on the TLS certs and chain
+// Process validates a request and sets context parameters according to the
+// configured AuthorizationCheckers.
+// Returns an http.Request with additional context values applied, or an error if
+// something went wrong.
+// In practice, this just calls ProcessWithParams.
+func (a *Auth) Process(w http.ResponseWriter, r *http.Request) (*http.Request, error) {
+	return a.ProcessWithParams(w, r, nil)
+}
+
+// ProcessWithParams validates a request and sets context parameters according to the
+// configured AuthorizationCheckers.
+// Returns an http.Request with additional context values applied, or an error if
+// something went wrong.
+func (a *Auth) ProcessWithParams(
+	w http.ResponseWriter, r *http.Request, ps httprouter.Params,
```

Comment:
The httprouter.Params come from the request context anyway, FWIW.

Reply:
This is true, but they are already parsed by `httprouter` and I wanted to avoid the extra effort since it's not required.